packages/core/documentation/src/components/DocumentationContent.vue

```diff
@@ -0,0 +1,174 @@
+            @child-change="(data) => emit('child-change', data)"
+            @selected="(data) => emit('document-selection', data)"
+          />
+          <DocumentationDisplay
+            v-if="selectedDocument"
+            :key="key"
+            :can-edit="canEdit"
+            class="document-holder"
+            :hide-publish-toggle="hidePublishToggle"
+            :is-card="isCard"
+            :selected-document="selectedDocument"
+            @add-clicked="handleAddClick"
+            @edit-clicked="handleEditClick"
+            @toggle-published="(data) => emit('toggle-published', data)"
+          />
+        </div>
+      </template>
+    </KCard>
+    <ProductDocumentModal
+      v-if="displayModal"
+      :documents="documentList"
+      :error-message="modalErrorMessage"
+      :hide-publish-toggle="hidePublishToggle"
+      :is-editing="isEditing"
+      :record="isEditing && selectedDocument ? selectedDocument : undefined"
+      @canceled="handleModalClosed"
+      @delete="emit('delete')"
+      @save="(formData: FormData, selectedFile: any) => emit('save', formData, selectedFile)"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, ref, watch } from 'vue'
+import DocumentationDisplay from './DocumentationDisplay.vue'
+import DocumentationPageEmptyState from './DocumentationPageEmptyState.vue'
+import ProductDocumentModal from './ProductDocumentModal.vue'
+import type { PropType } from 'vue'
+import type { DocumentListItem, DocumentTree, FormData } from '../types'
+import type { ChangeEvent, ChildChangeEvent, TreeListItem } from '@kong/kongponents'
+
+const emit = defineEmits<{
+  (e: 'child-change', data: ChildChangeEvent): void,
+  (e: 'delete'): void,
+  (e: 'document-selection', data: TreeListItem): void,
+  (e: 'modal-closed'): void,
+  (e: 'parent-change', data: ChangeEvent): void,
+  (e: 'save', formData: FormData, selectedFile: any): void,
+  (e: 'toggle-published', data: boolean): void,
+}>()
+
+const displayModal = ref<boolean>(false)
+const isEditing = ref<boolean>(false)
+
+const props = defineProps({
+  actionSuccess: {
+    type: Boolean,
+    default: false,
+  },
+  cacheKey: {
+    type: Number,
+    default: 0,
+  },
+  /**
+   * A synchronous or asynchronous function which returns a boolean evaluating
+   * if the user can edit an entity by create a new document
+   */
+  canEdit: {
+    type: Function as PropType<() => boolean | Promise<boolean>>,
+    required: false,
+    default: async () => false,
+  },
+  documentList: {
+    type: Array as PropType<DocumentListItem[]>,
+    required: true,
+  },
+  /**
+   * The ID of the entity to which a document is associated.
+   * Examples are API Product ID, Service ID etc.
+   */
+  entityId: {
+    type: String,
+    required: true,
+  },
+  hidePublishToggle: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+   * Boolean assiting with responsive documents view
+   */
+  isCard: {
```

Comment:
nit: we’re trying to move away from prefixed props - can this just be `card`

Reply:
fixed